packages/assets-controllers/src/TokensController.ts

```diff
@@ -624,12 +624,14 @@ export class TokensController extends BaseController<
       });
 
     this.update((state) => {
-      state.ignoredTokens = newIgnoredTokens;
-      state.tokens = newTokens;
       state.detectedTokens = newDetectedTokens;
```

Comment:
Do we want to move the setting of this property into the `if` as well?

Reply:
are you referring to the `detectedTokens` property? If so, I believe so. If the `interactingChainId` doesn't match the selected network, we should only update `allDetectedTokens`, as it corresponds to multichain functionality, whereas `detectedTokens` pertains specifically to the currently selected chain.

Reply:
Yes, that was my thinking exactly.

Reply:
done [aa7ded7](https://github.com/MetaMask/core/pull/5014/commits/aa7ded72cfe1fd4cda8388ff416119019f5f85eb)